File: algo/src/main/java/com/hartwig/actin/algo/evaluation/general/HasSufficientLifeExpectancy.java
package com.hartwig.actin.algo.evaluation.general;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class HasSufficientLifeExpectancy implements EvaluationFunction {

    HasSufficientLifeExpectancy() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        return Evaluation.IGNORED;
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/general/HasSufficientLifeExpectancy.kt
package com.hartwig.actin.algo.evaluation.general

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction
import com.hartwig.actin.datamodel.PatientRecord
import com.hartwig.actin.datamodel.algo.Evaluation

class HasSufficientLifeExpectancy : EvaluationFunction {
    
    override fun evaluate(record: PatientRecord): Evaluation {
        return EvaluationFactory.notEvaluated("Assumed that requested life expectancy will be met")
    }
}